sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1654,6 +1660,8 @@ object SQLConf {
       .doc("""The ID of session local timezone, e.g. "GMT", "America/Los_Angeles", etc.""")
       .version("")
       .stringConf
+      .checkValue(isValidTimezone, "The time zone should obey rules managed by" +
+        " java.time.zone.ZoneRulesProvider")
```

Comment:
This is still unclear, maybe just say: `Cannot resolve the give timezone with ZoneId.of(_, ZoneId.SHORT_IDS)`